### PR TITLE
Migrate Search Config Alias Grid

### PR DESCRIPTION
**Pull request by aleksiuno**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | This PR migrates "Shop parameters > Search > Search alias grid". It converts page to Symfony based. All other functionalities like alias add/edit forms alias toggling in the grid etc. will be implemented in future PR's as this one becomes large.
| Type?             | refacto
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | New grid needs to be displayed under `/configure/shop/search` route. Pagination and bulk delete features needs to be working.
| Fixed ticket?     | https://github.com/PrestaShop/PrestaShop/issues/32019
| Related PRs       | If theme, autoupgrade or other module change is needed to make this change work, provide a link to related PRs here.
| Sponsor company   | Your company or customer's name goes here (if applicable).

Before
image
After
image

### CODE REVIEW

File: src/PrestaShopBundle/Resources/config/services/core/grid/grid_definition_factory.yml [hunk 1]
```diff
@@ -386,3 +386,8 @@ services:
     parent: 'prestashop.core.grid.definition.factory.abstract_grid_definition'
     public: false
     autowire: true
+
+  prestashop.core.grid.definition.factory.alias:
+    class: 'PrestaShop\PrestaShop\Core\Grid\Definition\Factory\AliasGridDefinitionFactory'
```
Comment (zuk3975) [inline src/PrestaShopBundle/Resources/config/services/core/grid/grid_definition_factory.yml]: Use classname instead of custom id (and should be autowirable :thinking: )
```suggestion
  PrestaShop\PrestaShop\Core\Grid\Definition\Factory\AliasGridDefinitionFactory:
    autowire: true
```

File: src/PrestaShopBundle/Resources/config/services/core/grid/doctrine_query_builder.yml [hunk 2]
```diff
@@ -489,3 +489,10 @@ services:
       - '@prestashop.core.query.doctrine_search_criteria_applicator'
       - "@=service('prestashop.adapter.legacy.context').getContext().employee.id_lang"
     public: false
+
+  PrestaShop\PrestaShop\Core\Grid\Query\AliasQueryBuilder:
+    class: 'PrestaShop\PrestaShop\Core\Grid\Query\AliasQueryBuilder'
```
Comment (zuk3975) [inline src/PrestaShopBundle/Resources/config/services/core/grid/doctrine_query_builder.yml]:
```suggestion
```

File: src/PrestaShopBundle/Resources/config/routing/admin/configure/shop_parameters/search_preferences.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  path: /
+  methods: [ POST ]
+  defaults:
+    _controller: PrestaShopBundle\Controller\Admin\CommonController::searchGridAction
+    gridDefinitionFactoryServiceId: prestashop.core.grid.definition.factory.contacts
+    redirectRoute: admin_search_index
+    _legacy_controller: AdminSearchConf
+#    _legacy_link: AdminSearchConf:submitFiltercontact
+
+
```
Comment (zuk3975) [inline src/PrestaShopBundle/Resources/config/routing/admin/configure/shop_parameters/search_preferences.yml]:
```suggestion
```

File: src/PrestaShopBundle/Resources/config/routing/admin/configure/shop_parameters/search_preferences.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+admin_search_index:
+  path: /
+  methods: [ GET ]
+  defaults:
+    _controller: 'PrestaShopBundle\Controller\Admin\Configure\ShopParameters\SearchController::indexAction'
+    _legacy_controller: AdminSearchConf
+    # _legacy_link: AdminSearchConf
+    # _legacy_feature_flag: search
+
+admin_contacts_search:
```
Comment (zuk3975) [inline src/PrestaShopBundle/Resources/config/routing/admin/configure/shop_parameters/search_preferences.yml]:
```suggestion
admin_aliases_search:
```

Comment (zuk3975) [reply src/PrestaShopBundle/Resources/config/routing/admin/configure/shop_parameters/search_preferences.yml]: Im abit confused about naming too. Its abit unconventional page (but definatelly not contacts - thats a copy/paste leftover :smile: )

File: src/PrestaShopBundle/Resources/config/routing/admin/configure/shop_parameters/search_preferences.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  path: /
+  methods: [ POST ]
+  defaults:
+    _controller: PrestaShopBundle\Controller\Admin\CommonController::searchGridAction
+    gridDefinitionFactoryServiceId: prestashop.core.grid.definition.factory.contacts
```
Comment (zuk3975) [inline src/PrestaShopBundle/Resources/config/routing/admin/configure/shop_parameters/search_preferences.yml]: contacts?
Check other places too, probably left some copy paste mistakes (I assume you copied from contacts grid?)

File: src/PrestaShopBundle/Resources/config/routing/admin/configure/shop_parameters/search_preferences.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    redirectRoute: admin_search_index
+    _legacy_controller: AdminSearchConf
+#    _legacy_link: AdminSearchConf:submitFiltercontact
+
+
+admin_alias_delete_bulk:
```
Comment (zuk3975) [inline src/PrestaShopBundle/Resources/config/routing/admin/configure/shop_parameters/search_preferences.yml]:
```suggestion
admin_aliases_bulk_delete:
```

File: src/PrestaShopBundle/Resources/config/routing/admin/configure/shop_parameters/search_preferences.yml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  path: /delete/bulk
+  methods: [ POST ]
+  defaults:
+    _controller: 'PrestaShopBundle\Controller\Admin\Configure\ShopParameters\SearchController::deleteBulkAction'
```
Comment (zuk3975) [inline src/PrestaShopBundle/Resources/config/routing/admin/configure/shop_parameters/search_preferences.yml]:
```suggestion
    _controller: 'PrestaShopBundle\Controller\Admin\Configure\ShopParameters\SearchController::bulkDeleteAction'
```

File: src/Adapter/Search/AliasDeleter.php [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+namespace PrestaShop\PrestaShop\Adapter\Search;
+
+use Alias;
+use PrestaShopCollection;
+
+/**
+ * Class AliasDeleter deletes alias records, using legacy code.
+ */
+final class AliasDeleter
```
Comment (zuk3975) [inline src/Adapter/Search/AliasDeleter.php]: idk where you found this example but, delete it, its bad :D
And don't bother about deletion in this PR anyway. It will need to have dedicated CQRS command for deletion

File: src/PrestaShopBundle/Resources/config/services/adapter/alias.yml [hunk 9]
```diff
@@ -0,0 +1,6 @@
+services:
+  _defaults:
+    public: true
+
+  prestashop.adapter.alias.deleter:
+    class: 'PrestaShop\PrestaShop\Adapter\Search\AliasDeleter'
```
Comment (zuk3975) [inline src/PrestaShopBundle/Resources/config/services/adapter/alias.yml]: as mentioned bellow, you can delete this service and don't bother about deletion in this PR.

File: src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/SearchController.php [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+     *     message="You do not have permission to delete this."
+     * )
+     *
+     * @DemoRestricted(redirectRoute="admin_search_index")
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function deleteBulkAction(Request $request)
```
Comment (zuk3975) [inline src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/SearchController.php]: this can be removed as well as it uses wrong service. (you can move this part to a dedicated PR, where you would introduce DeleteAlias cqrs command and then integrate into list (when/if current PR is merged)

File: src/PrestaShopBundle/Resources/config/routing/admin/configure/shop_parameters/search_preferences.yml [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+  defaults:
+    _controller: PrestaShopBundle\Controller\Admin\CommonController::searchGridAction
+    gridDefinitionFactoryServiceId: prestashop.core.grid.definition.factory.contacts
+    redirectRoute: admin_search_index
+    _legacy_controller: AdminSearchConf
+#    _legacy_link: AdminSearchConf:submitFiltercontact
+
+
+admin_alias_delete_bulk:
+  path: /delete/bulk
```
Comment (zuk3975) [inline src/PrestaShopBundle/Resources/config/routing/admin/configure/shop_parameters/search_preferences.yml]:
```suggestion
  path: /bulk-delete
```

File: admin-dev/themes/new-theme/js/pages/search/SearchPage.ts [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+import SubmitRowActionExtension from '@components/grid/extension/action/row/submit-row-action-extension';
+import TranslatableInput from '@components/translatable-input';
+import ColumnTogglingExtension from '@components/grid/extension/column-toggling-extension';
+
+/**
+ * Responsible for actions in Contacts listing page.
+ */
+export default class SearchPage {
+  constructor() {
+    const AliasGrid = new Grid('alias');
```
Comment (zuk3975) [inline admin-dev/themes/new-theme/js/pages/search/SearchPage.ts]:
```suggestion
    const aliasGrid = new Grid('alias');
```

File: admin-dev/themes/new-theme/js/pages/search/SearchPage.ts [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+import SubmitBulkExtension from '@components/grid/extension/submit-bulk-action-extension';
+import BulkActionCheckboxExtension from '@components/grid/extension/bulk-action-checkbox-extension';
+import SubmitRowActionExtension from '@components/grid/extension/action/row/submit-row-action-extension';
+import TranslatableInput from '@components/translatable-input';
+import ColumnTogglingExtension from '@components/grid/extension/column-toggling-extension';
+
+/**
+ * Responsible for actions in Contacts listing page.
+ */
+export default class SearchPage {
```
Comment (zuk3975) [inline admin-dev/themes/new-theme/js/pages/search/SearchPage.ts]: this class is redundant, just move whole code into index.ts (that one represents the index page, don't need to wrap it up in any more layers)

File: src/Core/Grid/Query/AliasQueryBuilder.php [hunk 14]
```diff
@@ -0,0 +1,124 @@
...
+            'active' => 'a.active',
+        ];
+
+        foreach ($searchCriteria->getFilters() as $filterName => $filterValue) {
+            if (!array_key_exists($filterName, $allowedFiltersMap)) {
+                continue;
+            }
+
+            $builder
+                ->andWhere($allowedFiltersMap[$filterName] . ' LIKE :' . $filterName)
```
Comment (zuk3975) [inline src/Core/Grid/Query/AliasQueryBuilder.php]: id_alias field should be a strict search. (this is not a strict convention and we have some mixed behaviors, but I think thats the way we are going now - id fields filters uses strict search or a range input, but lets keep it simple for now.)
example here:
https://github.com/PrestaShop/PrestaShop/pull/31802/files#diff-4e2bb3656e21bd0da6e30a111a6dc99f42dd1dc487431714d80e7cfbff9a4f61R166

File: src/Core/Search/Filters/AliasFilters.php [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+     */
+    protected $filterId = AliasGridDefinitionFactory::GRID_ID;
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getDefaults(): array
+    {
+        return [
+            'limit' => 10,
```
Comment (zuk3975) [inline src/Core/Search/Filters/AliasFilters.php]: default value exists in parent class const
```suggestion
            'limit' => self::LIST_LIMIT,
```

File: src/Core/Grid/Query/AliasQueryBuilder.php [hunk 16]
```diff
@@ -117,7 +117,7 @@ private function applyFilters(QueryBuilder $builder, SearchCriteriaInterface $se
             }
 
             $builder
-                ->andWhere($allowedFiltersMap[$filterName] . ' LIKE :' . $filterName)
+                ->andWhere($allowedFiltersMap[$filterName] . ' = :' . $filterName)
```
Comment (zuk3975) [inline src/Core/Grid/Query/AliasQueryBuilder.php]: nope.
Only id_alias and active fields should be strict.
"alias" and "search" should stay with wildcard.

**Review by zuk3975 — APPROVED** [review]

**Review by FabienPapet — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]
Thanks @aleksiuno

**Comment by jolelievre** [comment]
Thanks @aleksiuno

This one can be merged since the page is still disabled and it's only a partial PR
